openshift_metrics/tests/test_metrics_processor.py

```diff
@@ -576,3 +576,74 @@ def test_condense_metrics_with_changing_gpu(self):
             ["cpu", "mem", "gpu_request", "gpu_type"]
         )
         self.assertEqual(condensed_dict, expected_condensed_dict)
+
+
+class TestExtractGPUInfo(TestCase):
+    def test_extract_gpu_info(self):
+        metric_with_label = {
+            "metric": {
+                "pod": "pod2",
+                "namespace": "namespace1",
+                "resource": "cpu",
+                "resource": "nvidia.com/gpu",
+                "label_nvidia_com_gpu_product": "V100-GPU",
+                "node": "node-1",
+                "label_nvidia_com_gpu_machine": "Dell PowerEdge",
+            },
+            "values": [
+                [60, 2],
+            ],
+        }
+
+        processor = metrics_processor.MetricsProcessor()
+        gpu_info = processor._extract_gpu_info("gpu_request", metric_with_label)
+
+        assert gpu_info.gpu_type == "V100-GPU"
+        assert gpu_info.gpu_resource == "nvidia.com/gpu"
+        assert gpu_info.node_model == "Dell PowerEdge"
+
+    def test_extract_gpu_info_with_missing_labels(self):
```

Comment:
Would it make sense to also add a test for the case where the the node is also not found in `gpu_node_map.json`?

Reply:
not sure how useful that would be but can't hurt so I added the test.